docs/hugo/content/contributing/add-a-new-code-generated-resource/create-a-sample.md

```diff
@@ -0,0 +1,37 @@
+---
+title: Create a sample to demonstrate the resource
+linktitle: Create a sample
+weight: 50
+---
+
+We mandate that every resource has a sample that demonstrates how to use it - and enforce this with checks in our CI pipeline. Good examples help other ASO users understand how to use each resource. 
+
+To ensure that each sample is correct, we run them - allowing us to catch any typos or errors before they trip up users in the wild.
+
+The samples are located in the [samples directory](https://github.com/Azure/azure-service-operator/blob/main/v2/samples). There should be at least one sample for each kind of supported resource. These currently need to be added manually. It's possible in the future we will automatically generate samples similar to how we automatically generate CRDs and types, but that doesn't happen today.
+
+The added new sample needs to be tested and recorded.
+
+If a recording for the test already exists, delete it. Look in the [recordings directory](https://github.com/Azure/azure-service-operator/blob/main/v2/internal/controllers/recordings/Test_Samples_CreationAndDeletion) for a file with the same name as your new test.  
+
+Typically these are named `Test_<GROUP>_<VERSION_PREFIX>_CreationAndDeletion.yaml`.
+For example, if we're adding sample for NetworkSecurityGroup resource, check for `Test_Network_v1beta_CreationAndDeletion.yaml`
+
+Run the test and record it using `task`:
+
+``` bash
+TEST_FILTER=Test_Samples_CreationAndDeletion task controller:test-integration-envtest
```

Comment:
Is there a way to run just the test for the sample that's been added rather than all of them?

Reply:
We use subtests so I think you can give the name of the subtest and it'll work?

It comes up a lot, we should document it - I often just comment the others out in the samples test code w/ a oneliner, but a cleaner way would be ideal to document.